javascript-carousel/main.js

```diff
@@ -0,0 +1,107 @@
+var $img = document.querySelector('img');
+var $circleSelectorAll = document.querySelectorAll('.circle');
+var $nextArrow = document.querySelector('.next-arrow');
+var $previousArrow = document.querySelector('.previous-arrow');
+
+var currentImage = 0;
+var imageSrcArray = ['images/001.png', 'images/004.png', 'images/007.png', 'images/025.png', 'images/039.png'];
+var timerId = null;
+
+function updatingPicture() {
+  if (currentImage === 4) {
+    currentImage = -1;
+  }
+  currentImage++;
+  for (let i = 0; i < imageSrcArray.length; i++) {
+    $img.setAttribute('src', imageSrcArray[currentImage]);
+    $img.setAttribute('data-image-index', `${currentImage}`);
+    for (let a = 0; a < $circleSelectorAll.length; a++) {
+      if (parseInt($circleSelectorAll[a].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+        $circleSelectorAll[a].className = 'circle circle-background';
+      } else {
+        $circleSelectorAll[a].className = 'circle';
+      }
+    }
+  }
+}
+
+function circleIndex(event) {
+  clearInterval(timerId);
+  timerId = null;
+  event.target.className = 'circle circle-background';
+  currentImage = parseInt(event.target.getAttribute('data-image-index'));
+  $img.setAttribute('src', imageSrcArray[parseInt(event.target.getAttribute('data-image-index'))]);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) !== parseInt(event.target.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
+  timerId = setInterval(updatingPicture, 3000);
+}
+
+function lastPicture(event) {
+  clearInterval(timerId);
+  if (currentImage === 0) {
+    currentImage = imageSrcArray.length - 1;
+    $img.setAttribute('src', imageSrcArray[currentImage]);
+    $img.setAttribute('data-image-index', currentImage);
+    for (let i = 0; i < $circleSelectorAll.length; i++) {
+      if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+        $circleSelectorAll[i].className = 'circle circle-background';
+      } else {
+        $circleSelectorAll[i].className = 'circle';
+      }
+    }
+    timerId = setInterval(updatingPicture, 3000);
+    return;
+  }
+  currentImage = parseInt($img.getAttribute('data-image-index')) - 1;
+  $img.setAttribute('src', imageSrcArray[currentImage]);
+  $img.setAttribute('data-image-index', parseInt($img.getAttribute('data-image-index')) - 1);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle circle-background';
+    } else {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
+  timerId = setInterval(updatingPicture, 3000);
+}
```

Comment:
This is pretty much the same code twice. If you update the current image and just roll with it, you don't need all of this code.